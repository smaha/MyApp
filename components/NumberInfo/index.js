import * as React from 'react';
import {
  View, Text, StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    flex: 0.5,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  label: {
    flex: 1,
    textAlign: 'center',
  },
  value: {
    flex: 1,
    fontSize: 20,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

const NumberInfo = ({ value, label }) => (
  <View style={styles.container}>
    { label && (
    <Text style={styles.label}>
      { label }
    </Text>
    )}
    <Text style={styles.value}>
      { value }
    </Text>
  </View>
);

NumberInfo.propTypes = {
  value: PropTypes.number,
  label: PropTypes.string,
};

export default NumberInfo;
