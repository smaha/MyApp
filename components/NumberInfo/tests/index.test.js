import React from 'react';
import { render } from '@testing-library/react-native';
import NumberInfo from '../index';

describe('<NumberInfo />', () => {
  test('To Match Snapshot', () => {
    const props = { value: 5, label: 'hello' };
    const rended = render(
      <NumberInfo {...props} />,
    );

    expect(rended.toJSON()).toMatchSnapshot();
  });
});
