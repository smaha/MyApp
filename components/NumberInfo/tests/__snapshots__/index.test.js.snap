// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInfo /> To Match Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 0.5,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "flex": 1,
        "textAlign": "center",
      }
    }
  >
    hello
  </Text>
  <Text
    style={
      Object {
        "flex": 1,
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    5
  </Text>
</View>
`;
